### PR TITLE
Added example for optimized full text search with GinIndex

### PR DESCRIPTION
**Pull request by roskakori**
This is a first proposal to improve the documentation on full text search performance.

For details, see <https://code.djangoproject.com/ticket/33459#ticket>.

### CODE REVIEW

File: docs/ref/contrib/postgres/search.txt [hunk 1]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
+    from django.contrib.postgres.search import SearchVectorField
+
+    class Entry(models.Model):
+        blog = models.ForeignKey(Blog, on_delete=models.CASCADE)
+        headline = models.CharField(max_length=255)
+        body_text = models.TextField()
+
+        # The search vector to store the content from the fields to actually
+        # search in an efficient and enriched format.
+        search_vector = models.SearchVectorField(null=True)
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:283]:
```suggestion
        search_vector = SearchVectorField(null=True)
```

File: docs/ref/contrib/postgres/search.txt [hunk 2]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
+
+    from django.contrib.postgres.search import SearchVectorField
+
+    class Entry(models.Model):
+        blog = models.ForeignKey(Blog, on_delete=models.CASCADE)
+        headline = models.CharField(max_length=255)
+        body_text = models.TextField()
+
+        # The search vector to store the content from the fields to actually
+        # search in an efficient and enriched format.
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:282]:
```suggestion
```

I would remove comments in code and move in the above text.

File: docs/ref/contrib/postgres/search.txt [hunk 3]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
+        search_vector = models.SearchVectorField(null=True)
+
+        class Meta:
+            # The index to search the search_vector efficiently.
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:286]:
```suggestion
```

I would remove comments in code and move in the above text.

File: docs/ref/contrib/postgres/search.txt [hunk 4]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
-model. You'll need to keep it populated with triggers, for example, as
-described in the `PostgreSQL documentation`_. You can then query the field as
-if it were an annotated ``SearchVector``::
+model and index it using a :class:`~django.contrib.postgres.indexes.GinIndex`
+or :class:`~django.contrib.postgres.indexes.GistIndex`.
 
-    >>> Entry.objects.update(search_vector=SearchVector('body_text'))
-    >>> Entry.objects.filter(search_vector='cheese')
+.. code-block:: python
+
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:273]: Some imports are missings

File: docs/ref/contrib/postgres/search.txt [hunk 5]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
+        # The search vector to store the content from the fields to actually
+        # search in an efficient and enriched format.
+        search_vector = models.SearchVectorField(null=True)
+
+        class Meta:
+            # The index to search the search_vector efficiently.
+            indexes = [GinIndex(fields=['search_vector']]
+
+Only once the other other fields have been stored in the database you can
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:289]:
```suggestion
Only once the other fields have been stored in the database you can
```

File: docs/ref/contrib/postgres/search.txt [hunk 6]
```diff
@@ -266,14 +266,54 @@ vector you wish to use. The PostgreSQL documentation has details on
...
+To use the search vector field for ranking, use an
+:class:`~django.db.models.F`: expression::
+
+    >>> from django.contrib.postgres.search import SearchQuery, SearchRank
+    >>> search_query = SearchQuery('cheese')
+    >>> search_rank = SearchRank(F('search_vector', search_query)
+    >>> Entry.objects.annotate(rank=search_rank).filter(
+    ...     search_vector=search_query
+    ... ).order_by('-rank')
+
```
Comment (pauloxnet) [inline docs/ref/contrib/postgres/search.txt:316]: Add also results as above.

**Review by pauloxnet — CHANGES_REQUESTED** [review]

**Comment by felixxm** [comment]
Closing per ticket.

**Pull request closed without merging**